lib/browser/api/dialog.js

```diff
@@ -70,147 +70,140 @@ const checkAppInitialized = function () {
   }
 }
 
-module.exports = {
-  showOpenDialog: function (...args) {
-    checkAppInitialized()
-
-    let [window, options, callback] = parseArgs(...args)
-
-    if (options == null) {
-      options = {
-        title: 'Open',
-        properties: ['openFile']
-      }
-    }
-
-    let {buttonLabel, defaultPath, filters, properties, title, message, securityScopedBookmarks = false} = options
-
-    if (properties == null) {
-      properties = ['openFile']
-    } else if (!Array.isArray(properties)) {
-      throw new TypeError('Properties must be an array')
-    }
+const setupDialog = (type, args) => {
+  let {
+    window,
+    buttonLabel,
+    defaultPath,
+    filters,
+    properties,
+    title,
+    message,
+    securityScopedBookmarks = false,
+    callback,
+    nameFieldLabel,
+    showsTagField
+  } = args
+
+  if (properties == null) {
+    properties = (type === 'open') ? ['openFile'] : ['saveFile']
+  } else if (!Array.isArray(properties)) {
+    throw new TypeError('Properties must be an array')
+  }
 
-    let dialogProperties = 0
+  let dialogProperties = 0
+  // set up dialogProperties for showOpenDialog
+  if (type === 'open') {
     for (const prop in fileDialogProperties) {
       if (properties.includes(prop)) {
         dialogProperties |= fileDialogProperties[prop]
       }
     }
+  }
 
-    if (title == null) {
-      title = ''
-    } else if (typeof title !== 'string') {
-      throw new TypeError('Title must be a string')
-    }
+  if (filters == null) filters = []
 
-    if (buttonLabel == null) {
-      buttonLabel = ''
-    } else if (typeof buttonLabel !== 'string') {
-      throw new TypeError('Button label must be a string')
-    }
+  if (title == null) {
+    title = ''
+  } else if (typeof title !== 'string') {
+    throw new TypeError('Title must be a string')
+  }
 
-    if (defaultPath == null) {
-      defaultPath = ''
-    } else if (typeof defaultPath !== 'string') {
-      throw new TypeError('Default path must be a string')
-    }
+  if (buttonLabel == null) {
+    buttonLabel = ''
+  } else if (typeof buttonLabel !== 'string') {
+    throw new TypeError('Button label must be a string')
+  }
+
+  if (defaultPath == null) {
+    defaultPath = ''
+  } else if (typeof defaultPath !== 'string') {
+    throw new TypeError('Default path must be a string')
+  }
+
+  if (message == null) {
+    message = ''
+  } else if (typeof message !== 'string') {
+    throw new TypeError('Message must be a string')
+  }
 
-    if (filters == null) {
-      filters = []
+  // set up extra parameters for showSaveDialog
+  if (type === 'save') {
```

Comment:
All possible `type` values should be defined in an enum, otherwise they are too typo-prone:

```js
const Type = {
  OPEN: 'open',
  SAVE: 'save'
}

// Later
if (type === Type.OPEN) {
  // Do some stuff.
}
```